gtsfm/loader/astronet_loader.py

```diff
@@ -0,0 +1,231 @@
+            img_fnames: file names of images in images_gtsfm
+            sfmtracks_gtsfm: tracks of points in points3D
+        """
+        # Note: Assumes input cameras use `PINHOLE` model
+        if len(images) == 0 and len(cameras) == 0:
+            raise RuntimeError("No Image or Camera data provided to loader.")
+        cameras_gtsfm, images_gtsfm, img_fnames = [], [], []
+        image_id_to_idx = {}  # keeps track of discrepencies between `image_id` and List index.
+        for idx, img in enumerate(images.values()):
+            images_gtsfm.append(Pose3(Rot3(img.qvec2rotmat()), img.tvec).inverse())
+            img_fnames.append(img.name)
+            fx, _, cx, cy = cameras[img.camera_id].params[:4]
+            cameras_gtsfm.append(Cal3Bundler(fx, 0.0, 0.0, cx, cy))
+            image_id_to_idx[img.id] = idx
+
+        sfmtracks_gtsfm = None
+        if len(points3D) > 0 and load_sfmtracks:
+            sfmtracks_gtsfm = []
+            for point3D in points3D.values():
+                sfmtrack = SfmTrack(point3D.xyz)
+                for (image_id, point2d_idx) in zip(point3D.image_ids, point3D.point2D_idxs):
+                    sfmtrack.add_measurement(image_id_to_idx[image_id], images[image_id].xys[point2d_idx])
+                sfmtracks_gtsfm.append(sfmtrack)
+
+        return cameras_gtsfm, images_gtsfm, img_fnames, sfmtracks_gtsfm
+
+    def __len__(self) -> int:
+        """The number of images in the dataset.
+
+        Returns:
+            the number of images.
+        """
+        return self._num_imgs
+
+    def get_image(self, index: int) -> Image:
+        """Get the image at the given index.
+
+        Args:
+            index: the index to fetch.
+
+        Raises:
+            IndexError: if an out-of-bounds image index is requested.
+
+        Returns:
+            Image: the image at the query index.
+        """
+        if index < 0 or index >= len(self):
+            raise IndexError(f"Image index {index} is invalid")
+
+        img = io_utils.load_image(self._image_paths[index])
+        return img
+
+    def get_camera_intrinsics(self, index: int) -> Cal3Bundler:
+        """Get the camera intrinsics at the given index.
+
+        Args:
+            the index to fetch.
+
+        Returns:
+            intrinsics for the given camera.
+        """
+        if index < 0 or index >= len(self):
+            raise IndexError(f"Image index {index} is invalid")
+        intrinsics = self._calibrations[index]
+        logger.info("Loading ground truth calibration.")
+
+        return intrinsics
+
+    def get_camera_pose(self, index: int) -> Optional[Pose3]:
+        """Get the camera pose (in world coordinates) at the given index.
+
+        Args:
+            index: the index to fetch.
+
+        Returns:
+            pose for the given camera.
+        """
+        if index < 0 or index >= len(self):
+            raise IndexError(f"Image index {index} is invalid")
+
+        if not self._use_gt_extrinsics:
```

Comment:
check if _wTi_list is None too?

Reply:
+1

Reply:
If `_wTi_list` is None then `_use_gt_extrinsics` is set to False in __init__.

Reply:
Added check. Should also be added to the other loaders